norfair/distances.py

```diff
@@ -345,110 +345,75 @@ def mean_manhattan(detection: "Detection", tracked_object: "TrackedObject") -> f
-
-    # Compute the intersection over union by taking the intersection
-    # area and dividing it by the sum of prediction + tracker
-    # areas - the interesection area
-    iou = inter_area / float(box_a_area + box_b_area - inter_area)
-    # Since 0 <= IoU <= 1, we define 1-IoU as a distance.
-    # Distance values will be in [0, 1]
-    return 1 - iou
-
-
-def iou(detection: "Detection", tracked_object: "TrackedObject") -> float:
-    """
-    Intersection over union distance between the bounding boxes.
 
-    Assumes that `detection.points` (and by consecuence `tracked_object.estimate`)
-    define a bounding box in the form `[[x0, y0], [x1, y1]]`.
+def iou(candidates: np.ndarray, objects: np.ndarray) -> np.ndarray:
+    """
+    Calculate IoU between two sets of bounding boxes. Both sets of boxes are expected
+    to be in `[x_min, y_min, x_max, y_max]` format.
 
     Normal IoU is 1 when the boxes are the same and 0 when they don't overlap,
     to transform that into a distance that makes sense we return `1 - iou`.
 
-    Performs checks that the bounding boxes are valid to give better error messages.
-    For a faster implementation without checks use [`iou_opt`][norfair.distances.iou_opt].
-
     Parameters
     ----------
-    detection : Detection
-        A detection.
-    tracked_object : TrackedObject
-        A tracked object.
+    candidates : numpy.ndarray
+        (N, 4) numpy.ndarray containing candidates bounding boxes.
+    objects : numpy.ndarray
+        (K, 4) numpy.ndarray containing objects bounding boxes.
 
     Returns
     -------
-    float
-        The distance.
+    numpy.ndarray
+        (N, K) numpy.ndarray of `1 - iou` between candidates and objects.
     """
-    boxa = detection.points.copy()
-    boxa.sort(axis=0)
-    _validate_bboxes(boxa)
-    boxb = tracked_object.estimate.copy()
-    boxb.sort(axis=0)
-    _validate_bboxes(boxb)
-    return _iou(boxa, boxb)
+    _validate_bboxes(candidates)
 
+    area_candidates = _boxes_area(candidates.T)
+    area_objects = _boxes_area(objects.T)
 
-def iou_opt(detection: "Detection", tracked_object: "TrackedObject") -> float:
-    """
-    Optimized version of [`iou`][norfair.distances.iou].
+    top_left = np.maximum(candidates[:, None, :2], objects[:, :2])
+    bottom_right = np.minimum(candidates[:, None, 2:], objects[:, 2:])
 
-    Performs faster but errors might be cryptic if the bounding boxes are not valid.
+    area_intersection = np.prod(
+        np.clip(bottom_right - top_left, a_min=0, a_max=None), 2
+    )
+    return 1 - area_intersection / (
+        area_candidates[:, None] + area_objects - area_intersection
+    )
 
-    Parameters
-    ----------
-    detection : Detection
-        A detection.
-    tracked_object : TrackedObject
-        A tracked object.
 
-    Returns
-    -------
-    float
-        The distance.
-    """
-    return _iou(detection.points, tracked_object.estimate)
+iou_opt = iou  # deprecated
```

Comment:
Is it still necessary with the change in line 415?

Reply:
I did this so that users can still import `iou_opt` as a function, trying not to make that many breaking changes.